flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/status/CommonStatus.java

```diff
@@ -93,4 +95,10 @@ public ResourceLifecycleState getLifecycleState() {
 
         return ResourceLifecycleState.DEPLOYED;
     }
+
+    /**
+     * Internal flag to signal that due to some condition we need to schedule a new reconciliation
+     * loop immediately. For example autoscaler overrides have changed and we need to apply them.
+     */
+    @JsonIgnore @Internal private boolean immediateReconciliationNeeded = false;
```

Comment:
This is the same behaviour we have for all upgrades. Rescheduling through the operator allows us to avoid potential deadlocks, have a fair reconciliation queue + to observe one more time before we execute a new action. This allows us to pick up any spec change that may have come in in the meantime.

Reply:
Currently the immediate reconciliation logic for upgrades is somewhat implicitly computed from the status. We could actually use this flag to make this more explicit later which helps understand the control flow.